test/e2e/util_gatewayapi_test.go

```diff
@@ -771,6 +782,50 @@ func assertCatalogSource(t *testing.T, namespace, csName string) error {
 	return err
 }
 
+func assertClusterServiceVersionSucceeded(t *testing.T) error {
+	t.Helper()
+	expectedVersion, err := expectedOSSMVersion(t)
```

Comment:
Would the starting CSV from the subscription do the same thing? We can return the starting CSV from `assertSubscription` function to avoid going down to the ingress operator's deployment.

Reply:
I used the value from the ingress operator deployment because it's the ultimate source of truth for what the operator is looking for, but you are right, the starting CSV reflects the same value, so it's equally valid.